### PR TITLE
Adds remindUser Endpoint

### LINKED ISSUE #175: [Feature Request] - Create supportRequest remind endpoint
### Pre-requisites:
- [x] I looked through both [open and closed issues](../issues?utf8=✓&q=is%3Aissue) and did not find another request for the same or similar feature.

## Is your feature request related to a problem, or an open issue? Describe the problem, and if applicable, reference the issue.

We need a new endpoint to remind users who have not come to start their support session

## Describe the solution you'd like

After popping a user off the queue, a support user/admin should be able to "remind the user" to head over for their support session. To support this, we need to create `/api/supportRequest/remindUser` which will accept a `supportRequestId` and a `relativeTimeElapsedString` (similar to `/abandonRequest`; use that as an example) and will need to utilize `messageUsers` to send a reminder to them. The reminder text should include the amount of time elapsed.

### PR DESCRIPTION
**Pull request by Ulfingrad**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

## Description of Changes

Maha and I added a remindUser endpoint which sends the user a message on Slack asking them to come to the booth (and includes the time elapsed since the support request was initially opened).
## Related Issues

Closes #175

### CODE REVIEW

**Comment by Ulfingrad** [comment]
Oops, testing for this is not quite finished! We will work on this once we have the chance to do so.

**Review by johnkahn — APPROVED** [review]

File: src/api/supportRequest.ts [hunk 1]
```diff
@@ -148,3 +148,24 @@ supportRequestRoutes.post('/abandonRequest', async (req, res) => {
...
+supportRequestRoutes.post('/remindUser', async (req, res) => {
+  const { supportRequestId, relativeTimeElapsedString } = req.body;
+  if (!supportRequestId || !relativeTimeElapsedString) {
+    res.status(400).send('One or more of the required properties is missing');
+    return;
+  }
+
+  try {
+    const supportRequest = await SupportRequest.findOne(supportRequestId);
+    await messageUsers(
```
Comment (SpencerKaiser) [inline src/api/supportRequest.ts:161]:
```suggestion
    const supportRequest = await SupportRequest.findOne(supportRequestId);
    if (!supportRequest) {
      res.status(404).send('Unable to find matching Support Request');
      return;
    }

    await messageUsers(
```

File: src/tests/api/supportRequest.test.ts [hunk 2]
```diff
@@ -157,6 +157,70 @@ describe('api/supportRequest', () => {
...
+    await supertest(app)
+      .post('/api/supportRequest/remindUser')
+      .send({ supportRequestId: 1, relativeTimeElapsedString: 'some time ago' })
+      .set({
+        Authorization: adminSecret,
+        'Content-Type': 'application/json',
+      })
+      .expect(500);
+  });
+
```
Comment (SpencerKaiser) [inline src/tests/api/supportRequest.test.ts:223]:
```suggestion

  it('will throw a 500 if the user cannot be messaged', async () => {
     jest.mock('../../slack/utilities/messageUsers');
     const { app } = require('../../app');
     await supertest(app)
       .post('/api/supportRequest/remindUser')
       .send({ supportRequestId: 1e6, relativeTimeElapsedString: 'some time ago' })
       .set({
         Authorization: adminSecret,
         'Content-Type': 'application/json',
       })
       .expect(404);
   });
```

File: src/tests/api/supportRequest.test.ts [hunk 3]
```diff
@@ -157,6 +157,82 @@ describe('api/supportRequest', () => {
...
+  it('will throw a 500 if the user cannot be messaged', async () => {
```
Comment (SpencerKaiser) [inline src/tests/api/supportRequest.test.ts:224]:
```suggestion
  it('will throw a 404 if the support request cannot be found', async () => {
```

**Review by SpencerKaiser — APPROVED** [review]

File: src/tests/api/supportRequest.test.ts [hunk 4]
```diff
@@ -157,6 +157,82 @@ describe('api/supportRequest', () => {
...
+     const { app } = require('../../app');
+     await supertest(app)
+       .post('/api/supportRequest/remindUser')
+       .send({ supportRequestId: 1e6, relativeTimeElapsedString: 'some time ago' })
+       .set({
+         Authorization: adminSecret,
+         'Content-Type': 'application/json',
+       })
+       .expect(404);
+   });
```
Comment (SpencerKaiser) [inline src/tests/api/supportRequest.test.ts:235]:
```suggestion
  });

```

Comment (SpencerKaiser) [inline src/tests/api/supportRequest.test.ts:235]:
```suggestion
    jest.mock('../../slack/utilities/messageUsers');
    const { app } = require('../../app');
    await supertest(app)
      .post('/api/supportRequest/remindUser')
      .send({ supportRequestId: 1e6, relativeTimeElapsedString: 'some time ago' })
      .set({
        Authorization: adminSecret,
        'Content-Type': 'application/json',
      })
      .expect(404);
  });
```

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]

**Comment by Ulfingrad** [comment]
I'm a bit busy today, but I will fix the new problems regarding this issue on Wednesday

**Review by SpencerKaiser — APPROVED** [review]

**Pull request merged**